### DISCUSSION TITLE
i am trying to get a list of all tables and their schema and their rules to display on the front end?

### DISCUSSION THREAD

**Discussion by gokulgseven** [pocketbase/pocketbase#4562] [Q&A]
this is what i have tried in extend as a framework as javascript 

```
const result = arrayOf(new DynamicModel({
    "id":     "",
    "system": "",
    "type":    "",
    "name":  "",
    "schema": [],
    "indexes": [],
    "listRule": "",
    "viewRule": "",
    "createRule": "",
    "updateRule": "",
    "deleteRule": "",
    "options": {},
    "created": "",
    "updated": "",
}))

$app.dao().db()
    .newQuery("SELECT id, system, type, name, schema, indexes, listRule, viewRule, createRule, updateRule,deleteRule, options, created, updated  FROM _collections")
    .all(result) 

console.log(JSON.stringify(result))
```

but i am getting the error 
`error GoError: sql: Scan error on column index 6, name "listRule": converting NULL to string is unsupported`

i found this api 
/api/collections

but for this i need the admin token

**Comment by gokulgseven** [comment]
$app.dao().db()
.newQuery("SELECT id, system, type, name, schema, indexes, COALESCE(listRule, ''), COALESCE(viewRule, ''), COALESCE(createRule, ''), COALESCE(updateRule, ''), COALESCE(deleteRule, ''), options, created, updated  FROM _collections")
.all(result) 
                
    found solution

**Comment by ganigeorgiev** [comment]
The reported error is caused because currently the dynamic model doesn't support nullable values (I'll consider it in the future).

Please don't just "prompt" and try to provide as much details as possible to understand the broader picture and your use case (see [The XY problem](https://xyproblem.info/)).

Your suggested solution is wrong because `null` rule has different semantic compared to empty string `""` (`null` means allowed only for admins; empty string means allowed for everyone).

In any case if you want to query the collection table the simplest solution is to replace your manually typed `const result = arrayOf(new DynamicModel())` with `const result = arrayOf(new Collection())`.

Or an alternative could be to use the collection helpers - https://pocketbase.io/docs/js-collections/. We don't have at the moment exposed method to fetch all collections but you can use the dao `collectionQuery()` helper to simplify your code:

```js
const collections = arrayOf(new Collection());
$app.dao().collectionQuery().all(collections)
```

**Reply by gokulgseven** [reply]
> Your suggested solution is wrong because null rule has different semantic compared to empty string "" (null means allowed only for admins; empty string means allowed for everyone).

i encountered an error further down the line due to this , spent 4 hours trying to figure out what was happening (couldnt solve it in the end) (will try your solution)

thanks
